### ISSUE TITLE
grouped.filter with axis=1 gave filtered rows

### ISSUE THREAD

**Issue by tdihp** [pydata/pandas#11041]
using pandas 0.16.2
minimum example here:

```
import numpy as np
import pandas as pd

index = pd.MultiIndex.from_product([range(10), [0, 1]])
data = pd.DataFrame(np.arange(100).reshape(-1, 20), columns=index)
data.groupby(level=0, axis=1).filter(lambda x: x.iloc[0, 0] > 10)
```

expected: get all filtered (grouped) columns as a DataFrame
result: `IndexError`

when change `np.arange(100)` to `np.arange(1000)` (more rows than columns)
results data frame with missing rows.

**Comment by jreback** [comment]
This is a bug [here](https://github.com/pydata/pandas/blob/master/pandas/core/groupby.py#L1209)

``self.axis`` needs to be passed to ``.take``

want to do a pull-request?

**Comment by terrytangyuan** [comment]
I'll submit a PR soon. But how to create this expected DataFrame in unit test? 
```
In [5]: data.groupby(level=0, axis=1).filter(lambda x: x.iloc[0, 0] > 10)
Out[5]:
    6       7       8       9
    0   1   0   1   0   1   0   1
0  12  13  14  15  16  17  18  19
1  32  33  34  35  36  37  38  39
2  52  53  54  55  56  57  58  59
3  72  73  74  75  76  77  78  79
4  92  93  94  95  96  97  98  99
```

**Comment by terrytangyuan** [comment]
So in R, there's this `dput()` method that we can use to get the declaration of a data frame. Is there a similar method in pandas that we can use to make the data frame reproducible?

**Comment by tdihp** [comment]
@terrytangyuan That's fast! :-P

**Comment by terrytangyuan** [comment]
@tdihp  thanks. Hope this could solve your problem.

**Comment by jreback** [comment]
closed by #11055
